src/merklized_state/data_source.rs

```diff
@@ -49,7 +49,7 @@ where
 {
     async fn get_path(
         &self,
-        state: &State,
+        tree_height: usize,
```

Comment:
Nit: I think this falls in the same category as the table and column names, where it is an implicit property of data stored in the database, and thus it must be fixed for any given instantiation of this API (or it requires a restart/migration to update). Thus it could probably also be a static method of `MerklizedState

Reply:
MerkleTreeScheme trait has a non static method for height so I was avoiding another method for height but it makes it cleaner so I'll add it